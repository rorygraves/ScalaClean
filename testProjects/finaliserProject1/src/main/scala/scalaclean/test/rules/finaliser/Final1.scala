package scalaclean.test.rules.finaliser.p1

object Final1 {
  // some final elements, but should not be marked as owner is already

  val a = 1
  def b = 2
  var c = 3
  a + b + c

}

class ShouldBeFinal {
  //not final as enclosing is
  def a = 1
  val b = 1
  var c = 1

  private class X
  private object Y
  class Z
}

abstract class NotFinalClass {
  def a = 1
  val b = 1
  var c = 1

  def fa = 1
  val fb = 1
  //vars are final
  var fc = 1

  def a1: Int
  val b1: Int
  var c1: Int

  private class X
  private object Y
  class Z
}

trait NotFinalTrait {
  def a = 1
  val b = 1
  var c = 1

  def a1: Int
  val b1: Int
  var c1: Int

  private class X
  private object Y
  class Z
}

class ExtNotFinalTrait extends NotFinalTrait {

  override def a1: Int = 1
  override val b1: Int = 1
  override var c1: Int = 1
}

class ExtNotFinalClass extends NotFinalClass {
  override def a = 1
  override val b = 1

  override def a1: Int = 1
  override val b1: Int = 1
  override var c1: Int = 1
}
