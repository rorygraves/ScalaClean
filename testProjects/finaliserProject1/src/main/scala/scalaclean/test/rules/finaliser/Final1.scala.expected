package scalaclean.test.rules.finaliser.p1

object Final1 {
  // some final elements, but should not be marked as owner is already

  val a = 1
  def b = 2
  var c = 3
  a + b + c

}

final class ShouldBeFinal {
  //not final as enclosing is
  def a = 1
  val b = 1
  var c = 1

  private class X
  private object Y
  final class Z
}

sealed abstract class NotFinalClass {
  def a = 1
  val b = 1
  var c = 1

  final def fa = 1
  final val fb = 1
  //vars are final
  var fc = 1

  def a1: Int
  val b1: Int
  var c1: Int

  private class X
  private object Y
  final class Z
}

sealed trait NotFinalTrait {
  final def a = 1
  final val b = 1
  var c = 1

  def a1: Int
  val b1: Int
  var c1: Int

  private class X
  private object Y
  final class Z
}

final class ExtNotFinalTrait extends NotFinalTrait {

  override def a1: Int = 1
  override val b1: Int = 1
  override var c1: Int = 1
}

final class ExtNotFinalClass extends NotFinalClass {
  override def a = 1
  override val b = 1

  override def a1: Int = 1
  override val b1: Int = 1
  override var c1: Int = 1
}
